simpleviewinc/keg-herkin
# Zen 604 copy clipboard

**Ticket**: https://jira.simpleviewtools.com/browse/ZEN-604

## Context

When the lines in a feature file extend beyond a certain boundary , there was additional styling and formatting required to the 'copy' option . It needed to be more user friendly , intuitive and placed correctly

## Goal

* Copy button should have a better styling
* The text from a line of the feature file should not disappear beyond the page when it crosses the column boundary
* The Copy button should remain in place , should not get cut off and should have an opaque background so that its easy to use

## Updates

* Updates to the component and theme of the definition list for both the styling and content
* *repos/tap/components/definition/definitionList.js*
  * The 'Copy to clipboard' text was considered too large and verbose . Changed as desired
* *repos/tap/theme/components/definitions/definitionList.js*
  * Changed the styling of the icon with text to appear as a flowing column rather than row
  * Colors , styling and text sizes according to expectation
  * set the copy feature to absolute and to the right so that it does not move and get covered up by the row text

## Testing

Docker image : docker pull ghcr.io/simpleviewinc/keg-herkin:zen-604-copyclipboard

* Steps
    * Pull this PR => keg herkin && keg pr 43
    * Navigate to repos/example folder and then start keg herkin => keg herkin start
    * Open launched app at http://herkin-zen-604-copyclipboard.local.keghub.io/?screen=editor&file=example.feature

* Test
  * When the window is active, Click or hover over any step in the 'List-Definitions' column on the right hand side
  * Confirm the Copy icon appears and is blue
  * Increase size by pressing 'Ctrl and +' to about 200 to 250%
  * ensure that some lines (active or inactive) that are beyond the column boundary overflow into the next lines and do not get cut off
  * Ensure that the Copy icon does not get cut off and always appears on the right hand side of an active row and is clearly legible
